src/org/labkey/test/components/domain/HitSelectionDialog.java

```diff
@@ -0,0 +1,93 @@
+package org.labkey.test.components.domain;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.components.ui.search.FilterExpressionPanel;
+import org.labkey.test.pages.LabKeyPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.labkey.test.components.html.Input.Input;
+
+
+public class HitSelectionDialog extends ModalDialog
+{
+
+    public HitSelectionDialog(WebDriver driver)
+    {
+        super(new ModalDialogFinder(driver));
+    }
+
+    public List<String> getAvailableFields()
+    {
+        return getWrapper().getTexts(elementCache().findFieldOptions());
+    }
+
+    public FilterExpressionPanel selectField(String fieldName)
+    {
+        var fieldItem = elementCache().findFieldOption(fieldName);
+        fieldItem.click();
+        return elementCache().filterExpressionPanel();
+    }
+
+    public void cancel()
+    {
+        dismiss("Cancel");
+    }
+
+    public void clickApply()
+    {
+        if (!elementCache().submitButton.isEnabled())
+        {
+            throw new IllegalStateException("Apply button is not enabled.");
+        }
+        elementCache().submitButton.click();
```

Comment:
Wait for dialog to close.
```suggestion
        elementCache().submitButton.click();
        waitForClose();
```
If possible, add some synchronization that waits for the `DomainFieldRow` to update.